### ISSUE TITLE
A11Y: Multiple issues with calendar widget

### ISSUE THREAD

**Issue by dreamofabear** [GoogleForCreators/web-stories-wp#4431]
## Bug Description

- When opening the widget, keyboard focus is not shifted from the main window to the calendar popup
- After opening the widget, keyboard focus is not trapped inside this calendar popup (can jump out to background)
- There is no way to dismiss the popup unless the user selects a date
- Previous/next buttons do not have appropriate labels and are read as "buttons"
- Minor: Dates don't announce the day of the week
- No state information provided to AMP/PM toggle buttons

## Screenshots

<img width="1004" alt="Screen Shot 2020-09-09 at 5 35 16 PM" src="https://user-images.githubusercontent.com/20525523/92657449-d9ceb000-f2c2-11ea-8981-e4501a499f78.png">

## Additional Context

- Plugin Version: 1.0.0-beta.2+fd68af7
- Operating System: macOS 10.15.6
- Browser: Chrome 85

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by dreamofabear** [comment]
Looks like issues with [react-calendar](https://github.com/wojtekmaj/react-calendar) so probably non-trivial to fix. 

We can probably push back on the priority of this, but curious if there aren't other 3P calendar widgets with better out-of-the-box A11Y. Added in #1743.

**Comment by bmattb** [comment]
@o-fernandez can you confirm that this should be prioritized by Prometheus for Stable in Sprint 37 right now

**Comment by barklund** [comment]
Until we implement our own calendar (icebox'ed in #1391) we can't really do anything about this. @o-fernandez to prioritize (and set target release).

**Comment by o-fernandez** [comment]
I don't *think* it will block stable release (@choumx to confirm) but let's prioritize for our upcoming sprint.

**Comment by barklund** [comment]
@o-fernandez, note that this probably requires prioritizing #1391 for next sprint too. But looking at it again, I'm not actually sure we can do this even after doing #1391.

@miina, can you do a really quick check whether the a11y requirements in this ticket are even possible with the current calendar component? Most are, but can we also change labels for prev/next buttons?

**Comment by miina** [comment]
Looked at the package a bit.

> When opening the widget, keyboard focus is not shifted from the main window to the calendar popup

The popup is managed by us and so is selecting the time part, we should be able to fix that.

> After opening the widget, keyboard focus is not trapped inside this calendar popup (can jump out to background)

Hmm, this seems to do with focus trapping to the selected element, e.g. using arrow keys (which theoretically should trigger changes inside the calendar) is moving the selected element instead. Probably need to adjust the focus trapping for the element.

> There is no way to dismiss the popup unless the user selects a date

That's also something we can add to the Popup.

> Previous/next buttons do not have appropriate labels and are read as "buttons"

It looks like `react-calendar` supports various labels as params, we should test out to be certain but looks like it's possible to add:
<img width="897" alt="Screenshot 2020-09-21 at 13 41 26" src="https://user-images.githubusercontent.com/3294597/93757903-cba85a00-fc07-11ea-915d-e6068d9b4b80.png">

> Minor: Dates don't announce the day of the week

The days of the week seem to have aria-labels currently. At the first glance doesn't seem to be possible to change any additional details but we could reach out to support if it's not working as expected.

> No state information provided to AMP/PM toggle buttons

That's our custom implementation, can be changed.

**Comment by csossi** [comment]
Verified in QA
